etc/tests/test_bootstrapper.py

```diff
@@ -0,0 +1,289 @@
+    notebook_file = os.getcwd() + "/etc/tests/resources/test-notebookA.ipynb"
+    notebook_output_html_file = "notebookA.html"
+    with tmpdir.as_cwd():
+        bootstrapper.convert_notebook_to_html(notebook_file, notebook_output_html_file)
+
+
+def test_fail_convert_notebook_to_html(tmpdir):
+    notebook_file = os.getcwd() + "/etc/tests/resources/test-bad-notebookA.ipynb"
+    notebook_output_html_file = "bad-notebookA.html"
+    with tmpdir.as_cwd():
+        with pytest.raises(nbformat.validator.NotebookValidationError):
+            bootstrapper.convert_notebook_to_html(notebook_file, notebook_output_html_file)
+
+
+def test_get_file_object_store(monkeypatch, s3_setup, tmpdir):
+    bucket_name = "test-bucket"
+    file_to_get = "test-file.txt"
+    monkeypatch.setattr(bootstrapper, "get_object_storage_filename", lambda x: file_to_get)
+
+    s3_setup.fput_object(bucket_name=bucket_name,
+                         object_name="test-file.txt",
+                         file_path="README.md")
+
+    with tmpdir.as_cwd():
+        bootstrapper.get_file_from_object_storage(s3_setup, bucket_name, file_to_get)
+
+
+def test_fail_get_file_object_store(monkeypatch, s3_setup, tmpdir):
+    bucket_name = "test-bucket"
+    file_to_get = "test-file.txt"
+    monkeypatch.setattr(bootstrapper, "get_object_storage_filename", lambda x: file_to_get)
+
+    with tmpdir.as_cwd():
+        with pytest.raises(minio.error.NoSuchKey):
+            bootstrapper.get_file_from_object_storage(s3_setup, bucket_name, file_to_get)
+
+
+def test_put_file_object_store(monkeypatch, s3_setup, tmpdir):
+    bucket_name = "test-bucket"
+    file_to_put = "LICENSE"
+    monkeypatch.setattr(bootstrapper, "get_object_storage_filename", lambda x: file_to_put)
+
+    bootstrapper.put_file_to_object_storage(s3_setup, bucket_name, file_to_put)
+
+    with tmpdir.as_cwd():
+        s3_setup.get_object(bucket_name, file_to_put)
+
+
+def test_fail_invalid_filename_put_file_object_store(monkeypatch, s3_setup):
+    bucket_name = "test-bucket"
+    file_to_put = "LICENSE_NOT_HERE"
+    monkeypatch.setattr(bootstrapper, "get_object_storage_filename", lambda x: file_to_put)
+
+    with pytest.raises(FileNotFoundError):
+        bootstrapper.put_file_to_object_storage(s3_setup, bucket_name, file_to_put)
+
+
+def test_fail_bucket_put_file_object_store(monkeypatch, s3_setup):
+    bucket_name = "test-bucket-not-exist"
+    file_to_put = "LICENSE"
+    monkeypatch.setattr(bootstrapper, "get_object_storage_filename", lambda x: file_to_put)
+
+    with pytest.raises(minio.error.NoSuchBucket):
+        bootstrapper.put_file_to_object_storage(s3_setup, bucket_name, file_to_put)
+
+
+def test_parse_arguments():
+    test_args = ['-e', 'http://test.me.now',
+                 '-d', 'test-directory',
+                 '-t', 'test-archive.tgz',
+                 '-i', 'test-notebook.ipynb',
+                 '-b', 'test-bucket']
+    bootstrapper.parse_arguments(test_args)
+
```

Comment:
Add confirmation.  Results should have keys in a dictionary- ensure "dest" keys exist.